src/librustc_codegen_ssa/base.rs

```diff
@@ -639,7 +638,18 @@ pub fn codegen_crate<B: ExtraBackendMethods>(
         ongoing_codegen.submit_pre_codegened_module_to_llvm(tcx, allocator_module);
     }
 
-    ongoing_codegen.submit_pre_codegened_module_to_llvm(tcx, metadata_module);
+    // njn: if I uncomment this condition I get tons of "can't find crate" test
+    // failures, like this:
+    //
+    // error[E0463]: can't find crate for `three_equals`
+    //   --> $DIR/three-equals.rs:6:1
+    //    |
+    //    | extern crate three_equals;
+    //    | ^^^^^^^^^^^^^^^^^^^^^^^^^^ can't find crate
+    //
+    //if tcx.sess.crate_types.borrow().contains(&config::CrateType::Dylib) {
```

Comment:
Oh you know I think I lied when I said that we only put compressed metadata in `Dylib` crate types. I think that we actually also put it in `ProcMacro` crate types, and that may fix the test error you're seeing?